dipy/denoise/localpca.py

```diff
@@ -0,0 +1,185 @@
+import numpy as np
+import scipy as sp
+import scipy.linalg as sla
+import scipy.special as sps
+from scipy.interpolate import interp1d
+from warnings import warn
+
+from dipy.core.ndindex import ndindex
+
+# Try to get the SVD through direct API to lapack:
+try:
+    from scipy.linalg.lapack import sgesvd as svd
+    svd_args = [1, 0]
+# If you have an older version of scipy, we fall back
+# on the standard scipy SVD API:
+except ImportError:
+    from scipy.linalg import svd
+    svd_args = [False]
+
+
+def inv_eta(phi):
+    """
+    The inverse of eta, which corrects for Rician bias at the end of the
+    denoising process (see eqs. 4,5,6 in [Manjon13]_).
+
+    Parameters
+    ----------
+    phi : float
+       signal to noise ratio
+
+    Returns
+    -------
+    inv_eta : float
+        The inverse of the eta function
+    """
+    phi_sq = phi ** 2
+    inv_eta1 = np.log(np.sqrt(np.pi/2)) - phi_sq / 2
+    inv_eta2 = 2 * np.log((((1 + (phi_sq / 2)) * sps.iv(0, phi_sq/4)) +
+                          ((phi_sq / 2) * sps.iv(1, (phi_sq/4)))))
```

Comment:
I was also using ive, which scales the bessel function by an exponential so it does not overflow, but then of course the exponential scaling back does overflow. I was about to wrote down the logs on a piece of paper to get them right, but you might want to look into it as you already got then written down.

Worse case SNR 50 something is the same as no correction at this point, as over 30 was doing nothing, and in pratice people don't bother changing it from 10. From my experience, it is negligible around SNR 30 already though.